internal/services/synapse/synapse_workspace_aad_admin_resource.go

```diff
@@ -139,6 +142,30 @@ func resourceSynapseWorkspaceAADAdminDelete(d *pluginsdk.ResourceData, meta inte
 		return err
 	}
 
+	workspace, err := workspaceClient.Get(ctx, id.ResourceGroup, id.WorkspaceName)
+	if err != nil {
+		if utils.ResponseWasNotFound(workspace.Response) {
+			return fmt.Errorf("retrieving %q: %+v", id, err)
+		}
+	}
+
+	if pointer.From(workspace.AzureADOnlyAuthentication) {
+		// Remove Azure Active Azure Admin, API requires AzureADOnlyAuthentication to be disabled first, see below error message from API.
```

Comment:
could we please add a test for this situation? thanks

Reply:
Thanks for your time. I assume that test case `TestAccSynapseWorkspaceAADAdmin_basic` has been covered this case(passed on my local) since I added `azuread_authentication_only = true` in the resource `azurerm_synapse_workspace` of template.